crates/turbopack-ecmascript/src/chunk/optimize.rs

```diff
@@ -267,6 +269,27 @@ async fn merge_duplicated_and_contained(
     Ok(())
 }
 
+async fn merge_by_size(chunks: &[EcmascriptChunkVc]) -> Result<Vec<EcmascriptChunkVc>> {
+    let mut merged = Vec::new();
+    let mut current = Vec::new();
+    let mut current_items = 0;
+    for &chunk in chunks {
+        let chunk_items = *chunk.chunk_items_count().await?;
+        if current_items == 0 || current_items + chunk_items < MAX_CHUNK_ITEMS_PER_CHUNK {
```

Comment:
One implies the other.
```suggestion
        if current_items + chunk_items < MAX_CHUNK_ITEMS_PER_CHUNK {
```

Reply:
It does not. `chunk_items` might be larger than MAX_CHUNK_ITEMS_PER_CHUNK

Reply:
Ohhh, I read this as `chunk_items`. Never mind.

Reply:
I made it a separate case to handle this case in a better way